velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -152,4 +152,29 @@ struct UnixTimestampParseWithFormatFunction
   bool invalidFormat_{false};
 };
 
+template <typename T>
+struct LastDayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE int64_t getYear(const std::tm& time) {
+    return 1900 + time.tm_year;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getMonth(const std::tm& time) {
+    return 1 + time.tm_mon;
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Date>& result,
+      const arg_type<Date>& date) {
+    auto dateTime = getDateTime(date);
+    int32_t year = getYear(dateTime);
+    int32_t month = getMonth(dateTime);
+    auto lastDay = util::getMaxDayOfMonth(year, month);
+    auto daysSinceEpoch = util::daysSinceEpochFromDate(year, month, lastDay);
```

Comment:
daysSinceEpochFromDate returns int64_t while Date constructor takes int32_t.
is there a possibility that daysSinceEpoch is > int32_t? I assume no?

Reply:
@laithsakka For daysSinceEpoch, int32_t is big enough, I also feel that daysSinceEpoch will not exceed the maximum value of int32_t.